### PR TITLE
fix: hotfix for redirect param in custom templates

### PR DESCRIPTION
**Pull request by TorstenDittmann**
## What does this PR do?

- fix redirect param not being set in custom templates
- dont escape redirect params in Mails worker

## Test Plan

- to be added

## Checklist

- [x] Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?
- [x] If the PR includes a change to an API's metadata (desc, label, params, etc.), does it also include updated API specs and example docs?

### CODE REVIEW

File: src/Appwrite/Platform/Workers/Mails.php [hunk 1]
```diff
@@ -67,7 +67,8 @@ public function action(Message $message, Registry $register): void
         $bodyTemplate = Template::fromFile($bodyTemplate);
         $bodyTemplate->setParam('{{body}}', $body, escapeHtml: false);
         foreach ($variables as $key => $value) {
-            $bodyTemplate->setParam('{{' . $key . '}}', $value);
+            // TODO: hotfix for redirect param
+            $bodyTemplate->setParam('{{' . $key . '}}', $value, escapeHtml: $key !== 'redirect');
```
Comment (stnguyen90) [inline src/Appwrite/Platform/Workers/Mails.php:71]: Why shouldn't we escape redirect?

File: app/controllers/api/account.php [hunk 2]
```diff
@@ -1069,6 +1068,7 @@
             /* {{user}} ,{{team}} and {{project}} are required in the templates */
```
Comment (christyjacob4) [inline app/controllers/api/account.php:1068]: lets add it back to the comment. and mention not to remove it

File: src/Appwrite/Platform/Workers/Mails.php (hunk 1 above)
Comment (TorstenDittmann) [reply src/Appwrite/Platform/Workers/Mails.php:71]: Bildschirmfoto 2024-01-11 um 20 43 46
source => https://www.php.net/manual/en/function.htmlspecialchars.php

File: app/controllers/api/account.php (hunk 2 above)
Comment (TorstenDittmann) [reply app/controllers/api/account.php:1068]: on it

File: src/Appwrite/Platform/Workers/Mails.php (hunk 1 above)
Comment (stnguyen90) [reply src/Appwrite/Platform/Workers/Mails.php:71]: `$redirect` should only be a URL and when the worker receives the `$bodyTemplate` it should either be a custom template or one of ours. Ours is the [email-inner-base.tpl](https://github.com/appwrite/appwrite/blob/bd920ead40b50b8be1cec32fcc8299ec2e5454a3/app/config/locale/templates/email-inner-base.tpl) with the following from the following replaced:

https://github.com/appwrite/appwrite/blob/b0a8aac0b7b7511523d8127792342bea9d552b85/app/config/locale/translations/en.json#L7-L11

For this example, it'd be:

```html
<p>Hey {{user}}</p>

<br>

<p>Follow this link to verify your email address.</p>

<a href="{{redirect}}" target="_blank">{{redirect}}</a>

<p>If you didn’t ask to verify this address, you can ignore this message.</p>

<br>

<p>Thanks</p>

<p>{{project}} team</p>
```

**Review by christyjacob4 — APPROVED** [review]

**Pull request merged**